tdp/cli/commands/deploy.py

```diff
@@ -99,16 +108,62 @@ def deploy(
         deployment_runner = DeploymentRunner(
             collections, ansible_executor, cluster_variables
         )
+        if resume:
+            if sources or targets:
+                raise click.BadParameter(
+                    f"sources or targets can not be used with resume"
+                )
+            resumed_deployment = get_deployment(session_class, resume)
+            if (
+                len(resumed_deployment.operations)
+                and resumed_deployment.operations[-1].state == StateEnum.SUCCESS
+            ):
+                raise click.BadParameter(
+                    f"Nothing to resume, deployment #{resume} was successful"
+                )
+            else:
+                click.echo(f"Resuming deployment #{resume}")
+            if resumed_deployment.targets is not None:
+                targets = resumed_deployment.targets
+            if resumed_deployment.sources is not None:
+                sources = resumed_deployment.sources
+            if resumed_deployment.filter_expression is not None:
+                filter = resumed_deployment.filter_expression
         if sources:
             click.echo(f"Deploying from {sources}")
         elif targets:
             click.echo(f"Deploying to {targets}")
         else:
             click.echo(f"Deploying TDP")
         try:
-            deployment_plan = DeploymentPlan.from_dag(
-                dag, sources=sources, targets=targets, filter_expression=filter
-            )
+            if resume:
+                deployment_plan_to_resume = DeploymentPlan.from_dag(
+                    dag, targets, sources, filter
+                )
+                original_operations = [
+                    operation.name for operation in deployment_plan_to_resume.operations
+                ]
+                # print(original_operations)
+                succeeded_operations = [
+                    operation.operation
+                    for operation in resumed_deployment.operations
+                    if operation.state == StateEnum.SUCCESS
+                ]
+                # print(succeeded_operations)
+                remaining_operations_list = list(
+                    set(original_operations) - set(succeeded_operations)
```

Comment:
This logic can not be moved in `DeploymentPlan` class ? We can add a method in `DeploymentPlan` which take a list of operations and change the plan with only remaining operations. Or maybe the method can take a `DeploymentLog` instance.